api/pipeline/v1/pipeline.proto

```diff
@@ -58,6 +58,9 @@ service PipelineService {
   // UpdatePipeline updates an existing pipeline and returns it.
   rpc UpdatePipeline (UpdatePipelineRequest) returns (Pipeline);
 
+  // CreateOrUpdatePipeline creates a new pipeline or updates an existing one and returns it.
+  rpc CreateOrUpdatePipeline (UpdatePipelineRequest) returns (Pipeline);
```

Comment:
Y'all likely have already had the discussion so I don't want to be a burden, but have we gotten over  `UpsertPipeline`?

I think it looks more inline with the rest here, WDYT?